### PR TITLE
refactor: replace experimental `maps` and `slices` with stdlib

### PR DESCRIPTION
**Pull request by Juneezee**
The experimental functions are now part of the standard library as of Go 1.21 and Go 1.23.

The key difference is that `maps.Keys` and `maps.Values` in the `golang.org/x/exp` package return a slice, whereas `maps.Keys` and `maps.Values` in the standard library return an iterator. To work with slices, we need to use `slices.Collect` to convert the iterator into a slice.

Reference: https://go.dev/doc/go1.21#slices
Reference: https://go.dev/doc/go1.23#iterators

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

### CODE REVIEW

**Comment by wozniakjan** [comment]
/run-e2e internal|opentelemetry_metrics

updated to latest `main` to check if this also works ok with https://github.com/kedacore/keda/pull/6378

**Comment by wozniakjan** [comment]
I thought it might happen that we will need to bump go back to 1.23

https://github.com/kedacore/keda/actions/runs/12155644728/job/33897625757?pr=6372

```
pkg/scalers/scalersconfig/typed_config.go:206:18: slices.Sorted requires go1.23 or later (file is go1.22)
pkg/scalers/scalersconfig/typed_config.go:206:30: maps.Keys requires go1.23 or later (file is go1.22)
pkg/scalers/scalersconfig/typed_config.go:464:21: slices.Sorted requires go1.23 or later (file is go1.22)
pkg/scalers/scalersconfig/typed_config.go:464:33: maps.Keys requires go1.23 or later (file is go1.22)
make: *** [Makefile:147: vet] Error 1
```

**Comment by Juneezee** [comment]
> I thought it might happen that we will need to bump go back to 1.23
> 
> https://github.com/kedacore/keda/actions/runs/12155644728/job/33897625757?pr=6372
> 
> ```
> pkg/scalers/scalersconfig/typed_config.go:206:18: slices.Sorted requires go1.23 or later (file is go1.22)
> pkg/scalers/scalersconfig/typed_config.go:206:30: maps.Keys requires go1.23 or later (file is go1.22)
> pkg/scalers/scalersconfig/typed_config.go:464:21: slices.Sorted requires go1.23 or later (file is go1.22)
> pkg/scalers/scalersconfig/typed_config.go:464:33: maps.Keys requires go1.23 or later (file is go1.22)
> make: *** [Makefile:147: vet] Error 1
> ```

@wozniakjan Thanks for correcting that for me :blush:

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,6 @@
 module github.com/kedacore/keda/v2
 
-go 1.22.1
-
-toolchain go1.23.3
+go 1.23.3
```
Comment (JorTurFer) [inline go.mod:3]: Please, revert this change. We decided to keep `go` version as the minimum required and defining `toolchain` with the required one -> https://github.com/kedacore/keda/pull/6378

**Review by zroubalik — COMMENTED** [review]
Looking good, just a minor nit

File: go.mod [hunk 2]
```diff
@@ -1,8 +1,6 @@
 module github.com/kedacore/keda/v2
 
-go 1.22.1
-
-toolchain go1.23.3
```
[the remark this answers is not in the document]
Comment (Juneezee) [reply go.mod:5]: image

When I add back the `toolchain` directive, `go mod tidy` removes it

[the remark this answers is not in the document]
Comment (zroubalik) [reply go.mod:5]: okay, let's keep it this way

**Review by zroubalik — APPROVED** [review]

**Comment by wozniakjan** [comment]
> This PR will revert the toolchain change of #6378 @wozniakjan , is it okey?

yeah, because at that point the `toolchain` and `go` versions are the same to support https://go.dev/doc/go1.23#iterators. I didn't realize this when approving https://github.com/kedacore/keda/pull/6378, but imho moving from experimental to core go packages has a higher value.

File: go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (wozniakjan) [reply go.mod:5]: yeah, it's because the versions exactly match. But I suspect we won't always be dependent on the cutting-edge go, so the `toolchain` will come back :)
